### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by Williamlee371**
5._get/api/reviews/:review_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -66,6 +72,40 @@ describe("Get", () => {
...
 		});
 	});
+	describe("api/reviews/:review_id", () => {
+		test("200-responses with a array with a single object", () => {
+			return request(app)
+				.get("/api/reviews/1")
+				.expect(200)
+				.then(({ body }) => {
+					expect(body["review"].length).toBe(1);
+					body["review"].forEach((review) => {
```
Comment (cjplatten) [inline __tests__/app.test.js]: review should be a single object, not an array.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -66,6 +72,40 @@ describe("Get", () => {
...
+				})
+			})
+		})
+		test('200-responses with an empty array if input a id that doesnt exist',()=>{
+			return request(app).get('/api/reviews/100').expect(200).then(({body})=>{
+				expect(body['review']).toEqual([])
+			})
+		})
+	});
 });
```
Comment (cjplatten) [inline __tests__/app.test.js:100]: Think about what errors you might encounter

File: controllers/controllers.js [hunk 3]
```diff
@@ -13,23 +13,37 @@ exports.getCategories = (request, response, next) => {
...
+};
+
+exports.getReviewsById = (request, response, next) => {
+	const {review_id} = request.params;
+	fetchReviewsById(review_id)
+		.then((review) => {
+			response.status(200).send({ review });
 		})
 		.catch((error) => {
 			next(error);
```
Comment (cjplatten) [inline controllers/controllers.js]: you might find it clearer to separate out you models and controllers by types, such as catergories and reviews

**Review by cjplatten — COMMENTED** [review]
Please see comments

File: __tests__/app.test.js [hunk 4]
```diff
@@ -66,15 +72,49 @@ describe("Get", () => {
 				});
 		});
 	});
+	describe("api/reviews/:review_id", () => {
+		test("200-responses with a single object", () => {
+			return request(app)
+				.get("/api/reviews/1")
+				.expect(200)
+				.then(({ body }) => {
```
Comment (sirKevlar) [inline __tests__/app.test.js:80]: Here we know which review we are getting back, so we could assert against actual values

File: __tests__/app.test.js [hunk 5]
```diff
@@ -66,15 +72,49 @@ describe("Get", () => {
...
+						});
+				});
+		});
+		test('200-responses with the correct item',()=>{
+			return request(app).get('/api/reviews/1').expect(200).then(({body})=>{
+					expect(body['review'].review_id).toBe(1)
+			})
+		})
+		test('200-responses with an empty array if input a id that doesnt exist',()=>{
+			return request(app).get('/api/reviews/100').expect(200).then(({body})=>{
```
Comment (sirKevlar) [inline __tests__/app.test.js]: This would be a status 404 for valid but non-existent id 'not found'

**Pull request merged**